core/src/main/java/pl/netigen/core/netigenapi/NetigenMainActivity.kt

```diff
@@ -2,104 +2,136 @@ package pl.netigen.core.netigenapi
 
 import android.os.Bundle
 import android.view.Gravity
+import android.view.View
 import android.widget.RelativeLayout
-import android.widget.Toast
 import androidx.annotation.CallSuper
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import com.google.ads.consent.ConsentInformation
 import com.google.android.gms.ads.AdSize
-import pl.netigen.billingandroid.IPaymentManager
-import pl.netigen.billingandroid.PaymentManager
-import pl.netigen.billingandroid.PurchaseListener
-import pl.netigen.core.R
 import pl.netigen.core.ads.AdsManager
 import pl.netigen.core.rewards.RewardsListener
-import java.lang.NullPointerException
+import pl.netigen.payments.IPaymentManager
+import pl.netigen.payments.PaymentManager
+import pl.netigen.payments.PurchaseListener
 
 abstract class NetigenMainActivity<ViewModel : NetigenViewModel> : AppCompatActivity(), PurchaseListener {
 
     open lateinit var viewModel: ViewModel
-    lateinit var paymentManager: IPaymentManager
+    private lateinit var paymentManager: IPaymentManager
     private var bannerRelativeLayout: RelativeLayout? = null
     var adsManager: AdsManager? = null
+    lateinit var consentInformation: ConsentInformation
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(getContentViewID())
         initPayments()
-        initAdsManager()
+        observeNoAds()
+        setConsentInformation()
+    }
+
+    private fun setConsentInformation() {
+        consentInformation = ConsentInformation.getInstance(this)
+        viewModel.isInEea = consentInformation.isRequestLocationInEeaOrUnknown
     }
 
     abstract fun getContentViewID(): Int
 
     private fun initPayments() {
         paymentManager = PaymentManager.createIPaymentManager(this)
-        paymentManager.isItemPurchased(viewModel.noAdsSku, this)
     }
 
-    private fun initAdsManager() {
-        if (!viewModel.isNoAdsBought) {
-            adsManager = AdsManager(viewModel, this)
-            if (viewModel.config.rewardedAdId != null) {
-                if (prepareRewardedAdsListener() == null)
-                    throw NullPointerException("Trying to load rewardedAds without a callback, prepareRewardedAdsListener should be overriden")
-                adsManager?.loadRewardedVideo()
+    private fun observeNoAds() {
+        viewModel.noAdsLiveData.observe(this, Observer {
+            if (it) {
```

Comment:
why not call onNoAdsBought()?

Reply:
Fixed with https://github.com/netigenkluzowicz/api_android/pull/74/commits/00c6bf8aa1d8a1d7c1e766d2b3e7f3c297c81308